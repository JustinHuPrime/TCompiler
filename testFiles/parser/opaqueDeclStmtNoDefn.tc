module foo;

void bar() {
  opaque o;
  o *ptr = null;
}