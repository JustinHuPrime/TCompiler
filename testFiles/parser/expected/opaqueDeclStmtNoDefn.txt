testFiles/parser/opaqueDeclStmtNoDefn.tc (code):
FILE(1, 1, STAB(ENTRY(bar, FUNCTION(testFiles/parser/opaqueDeclStmtNoDefn.tc, 3, 1, void()))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), FUNDEFN(3, 1, KEYWORDTYPE(3, 1, void), ID(3, 6, bar, REFERENCES(testFiles/parser/opaqueDeclStmtNoDefn.tc, 3, 1)), STAB(), COMPOUNDSTMT(3, 12, STAB(ENTRY(o, OPAQUE(testFiles/parser/opaqueDeclStmtNoDefn.tc, 4, 3, REFERENCES())), ENTRY(ptr, VARIABLE(testFiles/parser/opaqueDeclStmtNoDefn.tc, 5, 6, o *))), OPAQUEDECL(4, 3, ID(4, 10, o, REFERENCES(testFiles/parser/opaqueDeclStmtNoDefn.tc, 4, 3))), VARDEFNSTMT(5, 3, MODIFIEDTYPE(5, 3, POINTER, ID(5, 3, o, REFERENCES(testFiles/parser/opaqueDeclStmtNoDefn.tc, 4, 3))), ID(5, 6, ptr, REFERENCES(testFiles/parser/opaqueDeclStmtNoDefn.tc, 5, 6)), LITERAL(5, 12, NULL())))))
